guidance/models/_anthropic.py

```diff
@@ -52,24 +45,73 @@ def __init__(
 
     def _generator(self, prompt, temperature):
 
+        # find the role tags
+        pos = 0
+        role_end = b"<|im_end|>\n"
+        messages = []
+        found = True
+        system_prompt = None # Not mandatory, but we'll store it if found
+        while found:
```

Comment:
This block (or one very like it) seems to pop up a lot (I mentioned this when doing the AzureAI Studio examples). Can it be part of the chat template class (or similar)? After all, I believe that it's just 'undoing' the template.

Reply:
Of course it has to be more complicated :-(